src/input.rs

```diff
@@ -104,10 +107,21 @@ impl Input {
                 strobe_state: StrobeState{val: STROBE_STATE_A}
             },
             sdl: sdl,
+            _controllers: (0..num_joysticks().unwrap())
```

Comment:
Maybe this should rather be `.unwrap_or(0)`, but I'd imagine if this fails something is wrong